src/Visualization/Python/PlotAlongLine.py

```diff
@@ -0,0 +1,249 @@
+    "-s",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+    envvar="SPECTRE_MPL_STYLESHEET",
+    help=(
+        "Select a matplotlib stylesheet for customization of the plot, such "
+        "as linestyle cycles, linewidth, fontsize, legend, etc. "
+        "The stylesheet can also be set with the 'SPECTRE_MPL_STYLESHEET' "
+        "environment variable."
+    ),
+)
+def plot_along_line_command(
+    h5_files,
+    subfile_name,
+    list_vars,
+    vars,
+    step,
+    output,
+    line_start,
+    line_end,
+    num_threads,
+    stylesheet,
+):
+    """Plot variables along a line through volume data
+
+    Interpolates the volume data in the H5_FILES to a line and plots the
+    selected variables. You choose the line by specifying the start and end
+    points.
+    """
+    # Script should be a noop if input files are empty
+    if not h5_files:
+        return
+
+    # Open first H5 file to get some info
+    open_h5_file = spectre_h5.H5File(h5_files[0], "r")
+
+    # Print available subfile names and exit
+    if not subfile_name:
+        import rich.columns
+
+        available_subfiles = open_h5_file.all_vol_files()
+        if len(available_subfiles) == 1:
+            subfile_name = available_subfiles[0]
+        else:
+            rich.print(rich.columns.Columns(available_subfiles))
+            return
+
+    # Normalize subfile name
+    if subfile_name.endswith(".vol"):
+        subfile_name = subfile_name[:-4]
+    if not subfile_name.startswith("/"):
+        subfile_name = "/" + subfile_name
+
+    volfile = open_h5_file.get_vol(subfile_name)
+    obs_ids = volfile.list_observation_ids()
+    obs_values = list(map(volfile.get_observation_value, obs_ids))
+    dim = volfile.get_dimension()
+
+    # Select observation
+    if step is None:
+        raise click.UsageError(
+            f"Must specify '--step' (in [0, {len(obs_ids) - 1}], or -1)."
+        )
+    obs_id = obs_ids[step]
+
+    # Print available variables and exit
+    all_vars = volfile.list_tensor_components(obs_id)
+    if list_vars or not vars:
+        import rich.columns
+
+        rich.print(rich.columns.Columns(all_vars))
+        return
+    for var in vars:
+        if var not in all_vars:
+            raise click.UsageError(
+                f"Unknown variable '{var}'. Available variables are: {all_vars}"
+            )
+
+    # Close the H5 file because we're done with preprocessing
+    open_h5_file.close()
+
+    # Interpolate the selected quantities to the line
+    if line_start is None or line_end is None:
+        raise click.UsageError(
+            f"Specify '--line-start' / '-A' and '--line-end' / '-B'."
+        )
+    if len(line_start) != dim or len(line_end) != dim:
+        raise click.UsageError(
+            "Both '--line-start' / '-A' and '--line-end' / '-B' must have"
+            f" {dim} values (the dimension of the volume data)."
+        )
+    line_parameter = np.linspace(0, 1, 200)
```

Comment:
Should we allow the number of points to be an input? I don't really have a preference.

Reply:
Sure why not, added the parameter.